platform/darwin/src/MGLStyle.mm

```diff
@@ -627,10 +627,14 @@ - (void)localizeLabelsIntoLocale:(nullable NSLocale *)locale {
 
 - (NSArray<MGLStyleLayer *> *)placeStyleLayers {
     NSSet *streetsSourceIdentifiers = [self.mapboxStreetsSources valueForKey:@"identifier"];
-    
+
     NSSet *placeSourceLayerIdentifiers = [NSSet setWithObjects:@"marine_label", @"country_label", @"state_label", @"place_label", @"water_label", @"poi_label", @"rail_station_label", @"mountain_peak_label", @"natural_label", @"transit_stop_label", nil];
+    if (self.accessiblePlaceSourceLayerIdentifiers == nil) {
+        _accessiblePlaceSourceLayerIdentifiers = [NSMutableSet set];
+    }
+
     NSPredicate *isPlacePredicate = [NSPredicate predicateWithBlock:^BOOL (MGLVectorStyleLayer * _Nullable layer, NSDictionary<NSString *, id> * _Nullable bindings) {
-        return [layer isKindOfClass:[MGLVectorStyleLayer class]] && [streetsSourceIdentifiers containsObject:layer.sourceIdentifier] && [placeSourceLayerIdentifiers containsObject:layer.sourceLayerIdentifier];
+        return [layer isKindOfClass:[MGLVectorStyleLayer class]] && (([streetsSourceIdentifiers containsObject:layer.sourceIdentifier] && [placeSourceLayerIdentifiers containsObject:layer.sourceLayerIdentifier]) || [self.accessiblePlaceSourceLayerIdentifiers containsObject:layer.sourceLayerIdentifier]);
```

Comment:
Should we limit this additional check to only non-Streets sources? That would avoid accidentally enabling the feature for Streets layers that happen to have name collisions; on the other hand, it would prevent developers from potentially enabling the feature for more Streets layers we didn’t expect to be labeled but that the style happens to label.